Modifications for ease of use (#12)
elastic/cel-examples: README.md

Reviewer: A comment on a different file: `LICENSE`.
I think it'll need the `Copyright [yyyy] [name of copyright owner]` line to be filled out for Elastic.

### Overview
The cel-examples repository has a directory of fully functional examples as
well as a directory of FAQs. The goal of these resources is to help developers
working with the Elastic mito extension of CEL understand the language better
and to be more productive.

### CEL
Common Expression Language (CEL) is a non-Turing complete functional language.
Elastic chose CEL due to its performance and safety. Rather than using a sandbox
to run the program, CEL can be embedded as a library. Elastic extended CEL in
the mito project to include support for requesting and processing API requests.

> As a non-turing complete functional language, the structure of CEL programs can

Reviewer: ```suggestion
As a non-Turing complete functional language, the structure of CEL programs can
```

seem foreign to procedural or object-oriented programmers. For more information
on CEL and mito extensions, see the official documentation:
[Google CEL](https://github.com/google/CEL-spec/blob/master/doc/langdef.md),
[mito CEL extension](https://www.elastic.co/docs/reference/beats/filebeat/filebeat-input-cel).

### Examples
The examples directory has functioning CEL programs. Each example has a
docker-compose file to run a mock server to serve an API so the example can

> be run in realistically. The [README.md](./examples/README.md) explains how to

Reviewer: Now that's this readme.

[mito testdata](https://github.com/elastic/mito/tree/dev/testdata). The FAQ
code snippets are complementary to the code examples in the mito repository.
The code examples in the FAQs are focused on use in processing API requests.

## Structure of Examples Directory

Each example has its own directory with an agent configuration file 
(cel.yml.hbs file), a filebeat directory and a docker directory.
The displayed structure is based on using default values when
running '[beats-config-from-agent-config-template.sh](../scripts/beats-config-from-agent-config-template.sh)'
and running 'filebeat'.

```

> scripts
>     ├── beats-config-from-agent-config-template.sh   #script
>     ├── render.js                       #script that is used by 
>     │                                    beats-config-from-agent-config-template.sh
> └── filebeat-cel-input-base.yml     #File for the filebeat cel-input file to
>                                      which the CEL progam will be added.
>                                      Can be any name.  
> examples                                   
>     └──example_name/             # an example directory
>        ├── cel.yml.hbs           # agent configuration file. *
>        ├── variables.yml         # yaml file with replacement values for handlebar
>        │                           handlebar variables in the cel.yml.hbs file. * 
>        ├── filebeat/         # filebeat directory
>        │   ├── filebeat.yml  # filebeat executable configuration file *
>        │   ├── inputs.d      # generated directory to hold filebeat CEL input file *
>        │   │   └── test.yml  # generated filebeat config. 
>        │   └── out           # generated directory for output *
>        └── docker/           # docker files to run a test API server
>            ├── docker-compose.yml 
>            └── files
>                └── config.yml   # contains responses from the API Server

Reviewer: ```suggestion
.
└── scripts/
    ├── beats-config-from-agent-config-template.sh # Script.
    ├── render.js                                  # Helper used by the script.
    └── filebeat-cel-input-base.yml                # Filebeat CEL input config
                                                     to which the CEL progam
                                                     will be added. Can be any
                                                     name.
.
└── examples/
    └── example_name/              # An example directory.
        ├── cel.yml.hbs            # Agent configuration file. *
        ├── variables.yml          # YAML file with replacement values for
        │                            Handlebars variables in the agent
        │                            configuration file. *
        ├── filebeat/              # Filebeat directory.
        │   ├── filebeat.yml       # Filebeat executable configuration file. *
        │   ├── inputs.d/          # Generated directory for Filebeat config. *
        │   │   └── test.yml       # Generated Filebeat config.
        │   └── out/               # Generated directory for output. *
        └── docker/                # Docker files to run a test API server.
            ├── docker-compose.yml
            └── files/
                └── config.yml     # Contains responses from the API server.
```

* Consistency in space after hash at start of description.
* Correct indentation of filebeat-cel-input-base.yml.
* Remove trailing whitespace.
* Consistency in period at end of description.
* Wrap lines to 80 chars.
* Same indentation level for all `examples/` descriptions.
* Consistency in trailing slash for directory names
* Initial capital for descriptions.

Files with an "*" after the description can have their name and location
overridden. The locations must be in sync with the 'filebeat.yml' specified
on the commandline when running 'filebeat'. This is an advanced use and not 
expected in the normal flow of testing and debugging.                                   

> ```

Reviewer: This paragraph could be moved outside of the directory tree code block.
